integration/spark-common-test/src/test/scala/org/apache/carbondata/integration/spark/testsuite/complexType/TestAdaptiveEncodingForNullValues.scala

```diff
@@ -0,0 +1,168 @@
+  }
+
+
+  test("test SMALLINT with struct and array SMALLINT --> BYTE") {
+    sql("Drop table if exists adaptive")
+    sql(
+      "create table adaptive(roll int, student struct<id:smallint,name:string," +
+      "marks:array<smallint>>) stored by 'carbondata'")
+    sql("insert into adaptive values(1,'null$abc$null:null:null')")
+    checkAnswer(sql("select * from adaptive"),
+      Seq(Row(1, Row(null, "abc", mutable.WrappedArray.make(Array(null, null, null))))))
+  }
+
+
+  test("test BigInt with struct and array BIGINT --> BYTE") {
+    sql("Drop table if exists adaptive")
+    sql(
+      "create table adaptive(roll int, student struct<id:bigint,name:string," +
+      "marks:array<bigint>>) stored by 'carbondata'")
+    sql("insert into adaptive values(1,'null$abc$null:null:null')")
+    checkAnswer(sql("select * from adaptive"),
+      Seq(Row(1, Row(null, "abc", mutable.WrappedArray.make(Array(null, null, null))))))
+  }
+
+  test("test Double with Struct and Array DOUBLE --> BYTE") {
+    sql("Drop table if exists adaptive")
+    sql(
+      "create table adaptive(roll int, student struct<id:double,name:string," +
+      "marks:array<double>>) stored by 'carbondata'")
+    sql("insert into adaptive values(1,'null$abc$null:null:null')")
+    checkAnswer(sql("select * from adaptive"),
+      Seq(Row(1, Row(null, "abc", mutable.WrappedArray.make(Array(null, null, null))))))
+  }
+
+  test("test Decimal with Struct") {
+    sql("Drop table if exists adaptive")
+    sql(
+      "create table adaptive(roll int, student struct<id:decimal(3,2),name:string," +
+      "marks:array<decimal>>) stored by " +
+      "'carbondata'")
+    sql("insert into adaptive values(1,'null$abc$null:null:null')")
+    checkAnswer(sql("select * from adaptive"),
+      Seq(Row(1, Row(null, "abc", mutable.WrappedArray.make(Array(null, null, null))))))
+  }
+
+  test("test Timestamp with Struct") {
+    sql("Drop table if exists adaptive")
+    CarbonProperties.getInstance()
+      .addProperty(CarbonCommonConstants.CARBON_TIMESTAMP_FORMAT, "yyyy/MM/dd")
+    sql(
+      "create table adaptive(roll int, student struct<id:timestamp,name:string>) stored by " +
+      "'carbondata'")
+    sql("insert into adaptive values(1,'null$abc')")
+    checkAnswer(sql("select * from adaptive"),
+      Seq(Row(1, Row(null, "abc"))))
+  }
+
+  test("test Timestamp with Array") {
+    sql("Drop table if exists adaptive")
+    CarbonProperties.getInstance()
+      .addProperty(CarbonCommonConstants.CARBON_TIMESTAMP_FORMAT, "yyyy/MM/dd")
+    sql(
+      "create table adaptive(roll int, student struct<name:string," +
+      "marks:array<timestamp>>) stored by 'carbondata'")
+    sql("insert into adaptive values(1,'abc$null:null:null')")
+    checkAnswer(sql("select * from adaptive"),
+      Seq(Row(1, Row("abc", mutable.WrappedArray.make(Array(null, null, null))))))
+  }
+
+  test("test DATE with Array") {
```

Comment:
Write logic check it column is encoded as adaptive or not

Reply:
Will be done as part of CARBONDATA-2713